test/object-store/migrations.cpp

```diff
@@ -3032,280 +2258,278 @@ TEST_CASE("migration: AdditiveDiscovered") {
-            auto realm2 = Realm::get_shared_realm(config);
-            auto& group = realm2->read_group();
-            realm2->begin_transaction();
-            auto table = ObjectStore::table_for_object_type(group, "object");
-            auto col_keys = table->get_column_keys();
-            table->add_column(type_Int, "new column");
-            realm2->commit_transaction();
+        REQUIRE_NOTHROW(realm->update_schema(set_indexed(schema, "object", "value", false)));
+        REQUIRE(table->has_search_index(col_keys[0]));
 
-            REQUIRE_NOTHROW(realm->refresh());
-            REQUIRE(realm->schema() == schema);
-            REQUIRE(realm->schema().find("object")->persisted_properties[0].column_key == col_keys[0]);
-            REQUIRE(realm->schema().find("object")->persisted_properties[1].column_key == col_keys[1]);
-        }
-
-        DYNAMIC_SECTION("opening new Realms uses the correct schema after an external change" << mode_string) {
-            auto realm2 = Realm::get_shared_realm(config);
-            auto& group = realm2->read_group();
-            realm2->begin_transaction();
-            auto table = ObjectStore::table_for_object_type(group, "object");
-            auto col_keys = table->get_column_keys();
-            table->add_column(type_Double, "newcol");
-            realm2->commit_transaction();
+        REQUIRE_NOTHROW(realm->update_schema(set_indexed(schema, "object", "value 2", true)));
+        REQUIRE(!table->has_search_index(col_keys[1]));
+    }
 
-            REQUIRE_NOTHROW(realm->refresh());
-            REQUIRE(realm->schema() == schema);
-            REQUIRE(realm->schema().find("object")->persisted_properties[0].column_key == col_keys[0]);
-            REQUIRE(realm->schema().find("object")->persisted_properties[1].column_key == col_keys[1]);
+    SECTION("can remove properties from existing tables, but column is not removed") {
+        auto table = ObjectStore::table_for_object_type(realm->read_group(), "object");
+        REQUIRE_NOTHROW(realm->update_schema(remove_property(schema, "object", "value")));
+        REQUIRE(ObjectStore::table_for_object_type(realm->read_group(), "object")->get_column_count() == 2);
+        auto const& properties = realm->schema().find("object")->persisted_properties;
+        REQUIRE(properties.size() == 1);
+        auto col_keys = table->get_column_keys();
+        REQUIRE(col_keys.size() == 2);
+        REQUIRE(properties[0].column_key == col_keys[1]);
+    }
 
-            // Gets the schema from the RealmCoordinator
-            auto realm3 = Realm::get_shared_realm(config);
-            REQUIRE(realm->schema().find("object")->persisted_properties[0].column_key == col_keys[0]);
-            REQUIRE(realm->schema().find("object")->persisted_properties[1].column_key == col_keys[1]);
+    SECTION("cannot change existing property types") {
+        REQUIRE_THROWS_CONTAINING(realm->update_schema(set_type(schema, "object", "value", PropertyType::String)),
```

Comment:
This was previously trying to change to Float, which actually threw a schema validation error because the property is indexed.